rules/DowngradePhp81/Rector/FuncCall/DowngradeArrayIsListRector.php

```diff
@@ -147,12 +144,8 @@ private function shouldSkip(CallLike $callLike, ?Scope $scope): bool
             return true;
         }
 
-        if (! $scope instanceof Scope) {
-            $args = $callLike->getArgs();
-            return count($args) !== 1;
-        }
-
-        if ($scope->isInFunctionExists('array_is_list')) {
+        $scope = $callLike->getAttribute(AttributeKey::SCOPE);
```

Comment:
On after Unreachable Stmt, this Scope never exists, and will cause error.

The issue is not direct Unreachable, but after or even after that Stmt, and deep inside

Reply:
The scope is optional, so it's safe :+1:

Reply:
That required if that need to validate function_exists

Reply:
That's should be handled by `ReflectionProvider` already. Could you share a failing test fixture? I'd understand that better

Reply:
The PHPStan rely on Scope to now the position, with UnreachableNodeVisitor handling, Scope is filled, without that, no Scope at all as PHPStan Stop analyzing..., the existing downgrade fixtures already cover it

Reply:
ScopeAnalyzer is needed to pull Scope from deep Expr for this, eg:

```php
$$param = $$bar = $baz = array_is_list($value);
```

that covered at https://github.com/rectorphp/rector-src/blob/514df0a319b75f86b9b381ee0db7f95aadca32fb/src/NodeAnalyzer/ScopeAnalyzer.php#L50

that the `ScopeAnalyzer` to pull from current Stmt instead.